modules/021-cni-cilium/hooks/set_vxlan_allowed_ports.go

```diff
@@ -0,0 +1,178 @@
+/*
+Copyright 2021 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hooks
+
+import (
+	"strconv"
+
+	"github.com/flant/addon-operator/pkg/module_manager/go_hook"
+	"github.com/flant/addon-operator/sdk"
+	"github.com/flant/shell-operator/pkg/kube_events_manager/types"
+	appv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/utils/pointer"
+
+	"github.com/deckhouse/deckhouse/go_lib/set"
+)
+
+type Installation int
+
+const (
+	Existing Installation = iota
+	New
+)
+
+type Virtualization int
+
+const (
+	On Virtualization = iota
+	Off
+)
+
+var transitions = map[Installation]map[Virtualization][]TransitionRule{
+	Existing: { // existing installation
+		On: { // Virtualization is on
+			TransitionRule{source: 8469, target: 8469},
+			TransitionRule{source: 0, target: 8469},
+			TransitionRule{source: 8472, target: 4298},
+		},
+		Off: { // Virtualization is off
+			TransitionRule{source: 0, target: 8472},
+			TransitionRule{source: 8472, target: 8472},
+		},
+	},
+	New: { // new installation (DaemonSet "agent" does not exist)
+		On: { // Virtualization is on
+			TransitionRule{source: 0, target: 4298},
+			TransitionRule{source: 4298, target: 4298},
+			TransitionRule{source: 4299, target: 4298},
+		},
+		Off: { // Virtualization is off
+			TransitionRule{source: 0, target: 4299},
+			TransitionRule{source: 4299, target: 4299},
+		},
+	},
+}
+
+var _ = sdk.RegisterFunc(&go_hook.HookConfig{
+	OnBeforeHelm: &go_hook.OrderedConfig{Order: 10},
+	Queue:        "/modules/cni-cilium/set-vxlan-allowed-ports",
+	Kubernetes: []go_hook.KubernetesConfig{
+		{
+			Name:       "cilium-configmap",
+			ApiVersion: "v1",
+			Kind:       "ConfigMap",
+			NameSelector: &types.NameSelector{
+				MatchNames: []string{"cilium-config"},
+			},
+			NamespaceSelector: &types.NamespaceSelector{
+				NameSelector: &types.NameSelector{
+					MatchNames: []string{"d8-cni-cilium"},
+				},
+			},
+			ExecuteHookOnEvents:          pointer.Bool(false),
+			ExecuteHookOnSynchronization: pointer.Bool(false),
+			FilterFunc:                   filterConfigMap,
+		},
+		{
+			Name:       "agent-daemonset",
```

Comment:
Why don't you use the cilium-configmap to discover if cilium was deployed before?